cdap-app-fabric/src/main/java/co/cask/cdap/metadata/MetadataHttpHandler.java

```diff
@@ -80,737 +76,201 @@
+                             @QueryParam("sort") @DefaultValue("") String sort,
+                             @QueryParam("offset") @DefaultValue("0") int offset,
+                             // 2147483647 is Integer.MAX_VALUE
+                             @QueryParam("limit") @DefaultValue("2147483647") int limit,
+                             @QueryParam("numCursors") @DefaultValue("0") int numCursors,
+                             @QueryParam("cursor") @DefaultValue("") String cursor,
+                             @QueryParam("showHidden") @DefaultValue("false") boolean showHidden,
+                             @Nullable @QueryParam("entityScope") String entityScope) throws Exception {
+    if (searchQuery == null || searchQuery.isEmpty()) {
+      throw new BadRequestException("query is not specified");
+    }
+    Set<EntityTypeSimpleName> types = Collections.emptySet();
+    if (targets != null) {
+      types = ImmutableSet.copyOf(targets.stream().map(STRING_TO_TARGET_TYPE).collect(Collectors.toList()));
+    }
+    SortInfo sortInfo = SortInfo.of(URLDecoder.decode(sort, "UTF-8"));
+    if (SortInfo.DEFAULT.equals(sortInfo)) {
+      if (!(cursor.isEmpty()) || 0 != numCursors) {
+        throw new BadRequestException("Cursors are not supported when sort info is not specified.");
+      }
+    }
+    try {
+      MetadataSearchResponse response =
+        metadataAdmin.search(namespaceId, URLDecoder.decode(searchQuery, "UTF-8"), types,
+                             sortInfo, offset, limit, numCursors, cursor, showHidden, validateEntityScope(entityScope));
+      responder.sendJson(HttpResponseStatus.OK, GSON.toJson(response, MetadataSearchResponse.class));
+    } catch (Exception e) {
+      // if MetadataDataset throws an exception, it gets wrapped
+      if (Throwables.getRootCause(e) instanceof BadRequestException) {
+        throw new BadRequestException(e.getMessage(), e);
+      }
+      throw e;
+    }
   }
 
-  @DELETE
-  @Path("/namespaces/{namespace-id}/streams/{stream-id}/views/{view-id}/metadata/tags/{tag}")
-  public void removeViewTag(HttpRequest request, HttpResponder responder,
-                              @PathParam("namespace-id") String namespaceId,
-                              @PathParam("stream-id") String streamId,
-                              @PathParam("view-id") String viewId,
-                              @PathParam("tag") String tag) throws NotFoundException {
-    StreamViewId view = new StreamViewId(namespaceId, streamId, viewId);
-    metadataAdmin.removeTags(view, tag);
-    responder.sendString(HttpResponseStatus.OK,
-                         String.format("Tag %s for view %s deleted successfully.", tag, view));
+  private MetadataEntity getMetadataEntityFromPath(String uri, String s) {
+    String[] parts = uri.substring((uri.indexOf(Constants.Gateway.API_VERSION_3) +
+      Constants.Gateway.API_VERSION_3.length() + 1), uri.lastIndexOf(s)).split("/");
```

Comment:
does the uri contain only the path of the request, or also the query? The query might contain a match for the s.
Also please rename s to suffix.

Reply:
Thanks for renaming to suffix. 
I don't think you answered my question.